### PR TITLE
Feat(Query): Volume Has Sensitive Host Directory for Docker Compose

### PR DESCRIPTION
**Pull request by cxAndreFelicidade**
**Proposed Changes**
- Query: Volume Has Sensitive Host Directory (for Docker Compose)
- Functions: isOSDir, listcontains to common_lib (by Rogério Peixoto)

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: assets/libraries/common.rego [hunk 1]
```diff
@@ -744,3 +744,20 @@ remove_last_point(searchKey) = sk {
 } else = sk {
    sk := searchKey
 }
+
+isOSDir(mountPath) = result {
+	hostSensitiveDir = {
+		"/bin", "/sbin", "/boot", "/cdrom",
+		"/dev", "/etc", "/home", "/lib",
+		"/media", "/proc", "/root", "/run",
+		"/seLinux", "/srv", "/usr", "/var",
```
Comment (rafaela-soares) [inline assets/libraries/common.rego:753]: We should add "/sys" as recommended by AppSec

File: assets/queries/dockerCompose/volume_has_sensitive_host_directory/query.rego [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package Cx
+
+import data.generic.common as common_lib
+
+CxPolicy[result] {
+	resource := input.document[i]
+	service_parameters := resource.services[name]
+    volumes := service_parameters.volumes
+    host_path := split(volumes[0] ,":")
```
Comment (rafaela-soares) [inline assets/queries/dockerCompose/volume_has_sensitive_host_directory/query.rego:9]:
```suggestion
    volume := volumes[v]
    host_path := split(volume,":")
```

We want to check all the volumes and not the first one

File: assets/queries/dockerCompose/volume_has_sensitive_host_directory/query.rego [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    host_path1 := host_path[1]
+ 	common_lib.isOSDir(host_path1)
+    
+	result := {
+		"documentId": sprintf("%s", [resource.id]),
+		"searchKey": sprintf("services.%s.volumes",[name]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": "Docker compose file to have 'volumes' attribute not set to a sensitive directory",
+		"keyActualValue": "Docker compose file has 'volumes' attribute set to a sensitive directory",
```
Comment (rafaela-soares) [inline assets/queries/dockerCompose/volume_has_sensitive_host_directory/query.rego:18]:
```suggestion
		"keyExpectedValue": "There is no sensitive directory mounted as a volume",
		"keyActualValue": sprintf("There is a sensitive directory (%s) mounted as a volume", [hostPath1]),
		"searchLine": common_lib.build_search_line(["services", name, "volumes", v], []),
```

**Review by rafaela-soares — CHANGES_REQUESTED** [review]

File: assets/queries/dockerCompose/volume_has_sensitive_host_directory/query.rego [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package Cx
+
+import data.generic.common as common_lib
+
+CxPolicy[result] {
+	resource := input.document[i]
+	service_parameters := resource.services[name]
+    volumes := service_parameters.volumes
+    host_path := split(volumes[0] ,":")
+    host_path1 := host_path[1]
```
Comment (rafaela-soares) [inline assets/queries/dockerCompose/volume_has_sensitive_host_directory/query.rego:10]:
```suggestion
    host_path1 := host_path[0]
```

Please, update the samples. We should look to the left side

**Review by rafaela-soares — CHANGES_REQUESTED** [review]

File: assets/queries/dockerCompose/volume_has_sensitive_host_directory/query.rego [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+CxPolicy[result] {
+	resource := input.document[i]
+	service_parameters := resource.services[name]
+    volumes := service_parameters.volumes
+    volume := volumes[v]
+    source := volume.source
+    path := split(source,":")
+    host_path := path[0]
```
Comment (rafaela-soares) [inline assets/queries/dockerCompose/volume_has_sensitive_host_directory/query.rego:31]:
```suggestion
    host_path := volume.source
```

**Review by rafaela-soares — CHANGES_REQUESTED** [review]

File: assets/queries/dockerCompose/volume_has_sensitive_host_directory/query.rego [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+ 	common_lib.isOSDir(host_path)
+    
+	result := {
+		"documentId": sprintf("%s", [resource.id]),
+		"searchKey": sprintf("services.%s.volumes.source",[name]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": "There is no sensitive directory mounted as a volume",
+		"keyActualValue": sprintf("There is a sensitive directory (%s) mounted as a volume", [host_path]),
+		"searchLine": common_lib.build_search_line(["services", name, "volumes", "source"], []),
```
Comment (rafaela-soares) [inline assets/queries/dockerCompose/volume_has_sensitive_host_directory/query.rego:40]:
```suggestion
        "searchLine": common_lib.build_search_line(["services", name, "volumes", v, "source"], []),
```

**Review by rafaela-soares — CHANGES_REQUESTED** [review]

**Review by joaorufi — APPROVED** [review]

**Review by joaorufi — DISMISSED** [review]

**Review by joaorufi — APPROVED** [review]

**Review by rafaela-soares — APPROVED** [review]
LGTM 🚀

**Pull request merged**